### PR TITLE
Feature/sof 4405 - April Webinar

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: examples/job/adsorption-study.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        self.symbol = symbol
+        
+        # Get the job's material
+        material_id = self.job_json["_material"]["_id"]
+        self.material = self.material_endpoint.get(material_id)
+        
+    def __repr__(self):
+        return(str(self.name))
+        
+jobIds = (
```
[the remark this answers is not in the document]
Comment (AcylSilane) [reply examples/job/adsorption-study.py]: Changed it to run a query for the job set instead

File: examples/utils/material.py [hunk 2]
```diff
@@ -10,6 +10,25 @@
 
 from exabyte_api_client.endpoints.jobs import JobEndpoints
 
+def download_contcar(job_id: str, job_endpoint: JobEndpoints, filename: str):
```
Comment (timurbazhirov) [inline examples/utils/material.py:13]: This is an example of a non-modular implementation - let's define a `download_file_by_name` function, pls, and pass "CONTCAR" to it on invokation

File: examples/job/predict_with_machine_learning.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+
+
+
+
+# In[ ]:
+
+
+
+
```
Comment (timurbazhirov) [inline examples/job/predict_with_machine_learning.py:162]: Remove extra newlines

File: examples/job/predict_with_machine_learning.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+test_descriptors = descriptor_scaler.transform(test_descriptors)
+
+target_scaler = scaler()
+train_target = target_scaler.fit_transform(train_target)
+test_target = target_scaler.transform(test_target)
+
+
+# # Model Selection and Regression
+# 
+# Next, to demonstrate the programattic selection of models, we will choose between three types of model:
```
Comment (timurbazhirov) [inline examples/job/predict_with_machine_learning.py:65]: programmatic

File: examples/job/adsorption-study.py [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+    for job_id in job.job_ids:
+        job.job_endpoint.submit(job_id)
+
+
+# In[ ]:
+
+
+
+
```
Comment (timurbazhirov) [inline examples/job/adsorption-study.py:268]: Newlines

File: examples/job/adsorption-study.py [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+    adsorption_state.set_constraint(constraint)
+
+for job in jobs:
+    for adsorption_state in job.adsorption_states:
+        fix_all_slab_atoms(adsorption_state)
+
+
+# # Adsorption Study
+# 
+# Eads=Ecomplex-(Eslab+Emolecule)
```
Comment (timurbazhirov) [inline examples/job/adsorption-study.py:169]: This needs a more human-readable explanation

File: examples/job/adsorption-study.py [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+# 
+# Get slab, acquire science
+
+# In[]:
+
+
+job_endpoint = JobEndpoints(*ENDPOINT_ARGS)
+material_endpoint = MaterialEndpoints(*ENDPOINT_ARGS)
+
+# ToDo: Refactor class and add documentation
```
Comment (timurbazhirov) [inline examples/job/adsorption-study.py:43]: Let's actually do what this ToDo note is calling for.

File: examples/job/adsorption-study.py [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+
+# Import relevant portions of the API client
+from exabyte_api_client.endpoints.jobs import JobEndpoints
+from exabyte_api_client.endpoints.projects import ProjectEndpoints
+from exabyte_api_client.endpoints.materials import MaterialEndpoints
+from exabyte_api_client.endpoints.workflows import WorkflowEndpoints
+from exabyte_api_client.endpoints.bank_workflows import BankWorkflowEndpoints
+
+
+# # Query the Platform for the Slab
```
Comment (timurbazhirov) [inline examples/job/adsorption-study.py:33]: There should be an introductory statement at the top explaining what happened in the notebook and perhaps referring back to the webinar where it is explained (and the recording). See other examples.

File: examples/job/predict_with_machine_learning.py (hunk 3 above)
Comment (AcylSilane) [reply examples/job/predict_with_machine_learning.py:162]: Removed

File: examples/job/predict_with_machine_learning.py (hunk 4 above)
Comment (AcylSilane) [reply examples/job/predict_with_machine_learning.py:65]: Fixed

File: examples/job/adsorption-study.py (hunk 5 above)
Comment (AcylSilane) [reply examples/job/adsorption-study.py:268]: Removed

File: examples/job/adsorption-study.py (hunk 6 above)
Comment (AcylSilane) [reply examples/job/adsorption-study.py:169]: Expanded the explanation

File: examples/job/adsorption-study.py (hunk 8 above)
Comment (AcylSilane) [reply examples/job/adsorption-study.py:45]: Added explanations and a link to the webinar video

File: examples/utils/material.py (hunk 2 above)
Comment (AcylSilane) [reply examples/utils/material.py]: Defined a more general function as requested and changed download_contcar into a partial function of it

File: examples/job/adsorption-study.py (hunk 7 above)
Comment (AcylSilane) [reply examples/job/adsorption-study.py]: Added documentation for the class, removed the material attribute since it was un-used elsewhere

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**